patches/tModLoader/Terraria/ModLoader/ModPlayer.cs

```diff
@@ -340,6 +340,30 @@ public abstract class ModPlayer : ModType
 		public virtual void Kill(double damage, int hitDirection, bool pvp, PlayerDeathReason damageSource) {
 		}
 
+		/// <summary>
+		/// Called before vanilla makes any luck calculations. Return false to prevent vanilla from making their luck calculations. Returns true by default.
+		/// </summary>
+		/// <param name="luck"></param>
+		public virtual bool PreModifyLuck(ref float luck) {
+			return true;
+		}
+
+		/// <summary>
+		/// Allows you to modify a player's luck amount.
+		/// </summary>
+		/// <param name="luck"></param>
+		public virtual void ModifyLuck(ref float luck) {
+		}
+
+		/// <summary>
+		/// Allows you to modify the variables that affect overall torch luck. Increase positiveLuck if you want it to affect luck positively, and increase (note: increase) negativeLuck if you want to affect luck negatively.
+		/// </summary>
+		/// <param name="positiveLuck"></param>
+		/// <param name="negativeLuck"></param>
+		public virtual void ModifyTorchLuck(ref float positiveLuck, ref float negativeLuck) {
```

Comment:
This is really hard to use correctly. Please add much more info on exactly how `positiveLuck` and `negativeLuck` work.

Final `torchLuck` can only ever be 0.5 or 1, which results in either 0.1 or 0.2 luck bonus.
Tell modders that the normal use case is to "add 1 to positive luck for a good luck torch, or 1 to negative luck for a bad luck torch" "overall positiveLuck > 1 has no effect" "add a smaller amount (eg 0.5) for smaller impact, or add 2 to negativeLuck to negate all torch luck completely."